src/main/java/net/minestom/server/instance/Chunk.java

```diff
@@ -81,11 +84,15 @@
     // Data
     protected Data data;
 
+    // Generators
+    protected ChunkLightEngine chunkLightEngine;
```

Comment:
I don't think that we should allow different light engines per chunk. That seems like it would create very odd and undesirable behavior.

Reply:
> [This](https://www.seedofandromeda.com/blogs/29-fast-flood-fill-lighting-in-a-blocky-voxel-game-pt-1) is an extremely helpful article on how to do light propagation.

Thank you for the resource! I'll make sure to look into it once the API gets sorted out.

Reply:
Ah, I didn't notice changing the light engine per chunk. I'll make adjustments so it's just per-instance.